agent/engine/docker_client_test.go

```diff
@@ -361,18 +360,20 @@ func TestCreateContainerTimeout(t *testing.T) {
 	mockDocker, client, _, _, _, done := dockerClientSetup(t)
 	defer done()
 
-	warp := make(chan time.Time)
+	ctx, cancel := context.WithTimeout(context.TODO(), xContainerLongTimeout)
+	defer cancel()
```

Comment:
Since we expect the mock call to happen and we do a `cancel()` there, this `cancel()` can be removed.